### PR TITLE
Formalize supported browsers, drop transpilation, enforce with eslint

### PR DESCRIPTION
**Pull request by dannon**
Drops babel transpilation from build -- unnecessary tooling at this point.  Simplifies and speeds up the build.

To do this, this PR also formalizes supported browsers.  Explicitly drops support for IE11 (and below), which is EOL coincident with when this will make it to release (June 2022).

With this configuration there is now an additional ESlint plugin that will enforce feature (or polyfill) usage and warn when things are potentially problematic.

I also updated the pre-commit hook to use a small wrapper script running `eslint` appropriately for this use-case.  `pre-commit install` after updating your config to use it.

I refactored components which were using runtime vue template compilation (which we do not want to do anyway), and when doing so I noticed that many of our components didnt' adhere to the standard ordering.  I am following up in a separate branch to add eslint configuration to standardize this and apply it.

## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by dannon** [comment]
Problems with upload in selenium tests are probably real; I'll have to dig through what's going on there in the dependencies -- none of it got flagged in the language validation in eslint.

**Comment by bgruening** [comment]
:+1: for removing IE support and speeding up the build.

**Comment by bgruening** [comment]
Needs a rebase, otherwise this works for me locally.

**Comment by dannon** [comment]
@bgruening Thanks, I missed that it had conflicted!  Rebased now.

File: client/src/components/Upload/Collection.vue [hunk 1]
```diff
@@ -268,8 +268,13 @@ export default {
                     console.debug("Warning, upload response does not contain outputs.", model);
                 }
             });
-            const selection = new Galaxy.currHistoryPanel.collection.constructor(Object.values(models));
-            selection.historyId = Galaxy.currHistoryPanel.model.id;
+            // Build selection object
+            const selection = {
+                models: Object.values(models),
+                historyId: Galaxy.currHistoryPanel.model.id,
+                // Unwatch is unused?
```
Comment (bgruening) [inline client/src/components/Upload/Collection.vue]: It seems unused? Do you wanted to follow up here?

**Review by bgruening — APPROVED** [review]
LGTM. Is anyone __not__ ok with dropping IE support?

File: client/src/components/Upload/Collection.vue (hunk 1 above)
Comment (dannon) [reply client/src/components/Upload/Collection.vue]: I dug into it more; it's totally unused in this case and I'll just drop it here.  Thanks for the reminder!

**Pull request merged**